### PR TITLE
Fix for sched crash with entity limits

### PR DESCRIPTION
**Pull request by arungrover**
#### Describe Bug or Feature
Scheduler crashes when it is trying to check limits for an entity that has no other registered running job requesting the limit resource.

#### Describe Your Change
Change is simple, an 'if' condition was wrongly scoped. It ended up dereferencing uninitialized stack variables.

#### Link to Design Doc
NA

#### Attach Test and Valgrind Logs/Output
I ended up extending an existing limits smoketest.
[test_before.txt](https://github.com/PBSPro/pbspro/files/3790263/test_before.txt)
[test_after.txt](https://github.com/PBSPro/pbspro/files/3790264/test_after.txt)

### CODE REVIEW

File: src/scheduler/check.c [hunk 1]
```diff
@@ -1341,10 +1341,9 @@ find_counts_elm(counts *cts_list, char *name, resdef *rdef, counts **cnt, resour
...
 		if (rdef == NULL)
 			return cts->running;
-		else {
-			if ((res_lim = find_resource_count(cts->rescts, rdef)) != NULL)
-				if (rcount != NULL)
-					*rcount = res_lim;
+		else if ((res_lim = find_resource_count(cts->rescts, rdef)) != NULL) {
+			if (rcount != NULL)
+				*rcount = res_lim;
 			return res_lim->amount;
```
Comment (agrawalravi90) [inline src/scheduler/check.c]: So, I think this might be changing existing logic. Previously, the code would return res_lim->amount regardless of the if ((res_lim = find_resource_count(cts->rescts, rdef)) condition, now it will return it only if that condition is true, is this change intentional?

Comment (arungrover) [reply src/scheduler/check.c]: Right that's the bug, because, find_resource_count returns "res_lim". so unless it is not-null, we end up dereferencing an uninitialized pointer.
